### PR TITLE
Add CSI Windows Support Doc

### PR DESCRIPTION
**Pull request by jingxu97**
add doc for csi windows support

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,6 +64,10 @@ This driver supports only one topology key:
 `topology.gke.io/zone`
 that represents availability by zone (e.g. `us-central1-c`, etc.).
 
+### CSI Windows Support
```
Comment (msau42) [inline README.md:67]: Can the versioning information instead be put in the "Features In development" section below?

File: README.md [hunk 2]
```diff
@@ -64,6 +64,10 @@ This driver supports only one topology key:
...
+
+GCE PD driver starts to support CSI Windows (alpha) with version v1.0.*. It requires to install [CSI Proxy] (https://github.com/kubernetes-csi/csi-proxy) binary on each Windows node. The latest version of CSI Proxy is Beta. GCE PD driver start to support Beta CSI Proxy starting from v1.1.0. Starting Kubernetes 1.19, CSI Proxy binary is automatically installed during node start up. In Kubernetes 1.20, CSI Proxy will be running a windows service to improve its stablibity.
```
Comment (msau42) [inline README.md:69]: I don't think we can make a lot of statements about csi proxy installation. People may not be using kube-up or GKE.

We probably should say that csi-proxy needs to be installed by the cluster distribution/admin on all nodes, and mention that it is installed with kube-up, and plans to be installed automatically by GKE.

File: docs/kubernetes/user-guides/windows.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+$ GCE_PD_SA_NAME=my-gce-pd-csi-sa                 # Name of the service account to create
+$ GCE_PD_SA_DIR=/my/safe/credentials/directory    # Directory to save the service account key
+$ ./deploy/setup-project.sh
+```
+
+2. Deploy driver to Kubernetes Cluster
+
+```
+$ GCE_PD_SA_DIR=/my/safe/credentials/directory    # Directory to get the service account key
+$ GCE_PD_DRIVER_VERSION=alpha                     # Currently alpha deploy Driver version with Windows Alpha support. Will add beta supporot to beta GCE_PD_DRIVER_VERSION deployment script.
```
Comment (msau42) [inline docs/kubernetes/user-guides/windows.md:25]: Are we going to move this to the stable overlay?

File: docs/kubernetes/user-guides/windows.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# Kubernetes CSI Windowos User Guide
```
Comment (msau42) [inline docs/kubernetes/user-guides/windows.md:1]: Any other steps needed to use windows? Such as StorageClass settings? Can we add example Pod/pvc/storageclass specs?

File: docs/kubernetes/user-guides/windows.md (hunk 3 above)
Comment (jingxu97) [reply docs/kubernetes/user-guides/windows.md:25]: I will do this a little bit later.

File: docs/kubernetes/user-guides/windows.md (hunk 4 above)
Comment (jingxu97) [reply docs/kubernetes/user-guides/windows.md:1]: storage class for linux works just works for linux too.
 I will add example

**Review by jingxu97 — COMMENTED** [review]
@msau42 Updated the doc. PTAL

File: examples/demo-windows.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      storage: 1Gi
+
+---
+
+apiVersion: v1
+kind: Pod
+metadata:
+  name: web-windows
+spec:
+  tolerations:
```
Comment (msau42) [inline examples/demo-windows.yaml:20]: Is a blanket toleration a good idea? What about just tolerating the Windows taint? Was there not an admission webhook that automatically adds the toleration?

File: docs/kubernetes/user-guides/windows.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+$ ./deploy/setup-project.sh
+```
+
+2. Deploy driver to Kubernetes Cluster
+
+```
+$ GCE_PD_SA_DIR=/my/safe/credentials/directory    # Directory to get the service account key
+$ GCE_PD_DRIVER_VERSION=alpha                     # Currently alpha deploy Driver version with Windows Alpha support. Will add beta supporot to beta GCE_PD_DRIVER_VERSION deployment script.
+$ ./deploy/kubernetes/deploy-driver.sh
+```
```
Comment (msau42) [inline docs/kubernetes/user-guides/windows.md:29]: After install, link to basic.md to create Pod?

File: examples/demo-windows.yaml (hunk 5 above)
Comment (mattcary) [reply examples/demo-windows.yaml:20]: At any rate documenting the assumption about what windows taints are expected might be a good idea?

File: docs/kubernetes/user-guides/windows.md (hunk 6 above)
Comment (jingxu97) [reply docs/kubernetes/user-guides/windows.md:29]: updated it.

File: examples/demo-windows.yaml (hunk 5 above)
Comment (jingxu97) [reply examples/demo-windows.yaml:20]: In OSS, acccording to code https://github.com/kubernetes/kubernetes/search?q=WINDOWS_NODE_TAINTS, test cluster set "node.kubernetes.io/os=win1809:NoSchedule". I think this should be updated for different os windows cluster. follow up with an issue to track it.

In GKE, the related doc is here https://cloud.google.com/kubernetes-engine/docs/how-to/deploying-windows-app
It does not contain any os version information.

File: docs/kubernetes/user-guides/windows.md (hunk 4 above)
Comment (jingxu97) [reply docs/kubernetes/user-guides/windows.md:1]: added example

**Comment by jingxu97** [comment]
updated the PR. PTAL

File: README.md [hunk 7]
```diff
@@ -64,6 +64,10 @@ This driver supports only one topology key:
 `topology.gke.io/zone`
 that represents availability by zone (e.g. `us-central1-c`, etc.).
 
+### CSI Windows Support
+
+GCE PD driver starts to support CSI Windows with [CSI Proxy] (https://github.com/kubernetes-csi/csi-proxy). It requires to install csi-proxy.exe binary on each Windows node. Please see more details in CSI Windows page (docs/kubernetes/user-guides/windows.md)
```
Comment (msau42) [inline README.md:69]: "It requires csi-proxy.exe to be installed on every Windows node"

File: README.md [hunk 8]
```diff
@@ -72,7 +76,7 @@ that represents availability by zone (e.g. `us-central1-c`, etc.).
 | Snapshots       | Beta  | 1.17                          | Any                          | v1.0.0             | Stable             |
 | Resize (Expand) | Alpha | 1.14                          | 1.14                         | v0.6.0             | Alpha              |
 | Resize (Expand) | Beta  | 1.16                          | 1.16                         | v0.7.0             | Stable             |
-| Windows*        | Beta  | 1.18                          | 1.18                         | v1.1.0             | Stable             |
+| Windows*        | Beta  | 1.18                          | 1.18                         | v1.1.0             | Alpha             |
```
Comment (msau42) [inline README.md:79]: Is this available in stable now?

File: docs/kubernetes/user-guides/windows.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
+# Kubernetes CSI Windowos User Guide
```
Comment (msau42) [inline docs/kubernetes/user-guides/windows.md:1]: typo: Windows

File: docs/kubernetes/user-guides/windows.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+>**Attention:** CSI Windows Beta is only available in the driver version v1.1.0+
+>**Attention:** CSI Windows Alpha is only available in the driver version v1.0.0-v1.0.*
+
+### Install CSI Proxy binary
+CSI proxy can be installed as binary or run as a Windows service on each Windows node. Please see details on CSI Proxy [installation and usage page](https://github.com/kubernetes-csi/csi-proxy/blob/master/README.md#usage).
+
+If you are using kube-up to start a GCE Kubernetes cluster, starting Kubernetes 1.19, [CSI Proxy Alpha](https://github.com/kubernetes-csi/csi-proxy/releases/tag/v0.1.0) binary is automatically installed during node start up. In Kubernetes 1.20, [CSI Proxy Beta.1 v0.2.1](https://github.com/kubernetes-csi/csi-proxy/releases/tag/v0.2.1) is installed and running as a windows service to improve its stablibity. For GKE cluster, starting from 1.18, CSI proxy will be installed automatically.
```
Comment (msau42) [inline docs/kubernetes/user-guides/windows.md:9]: I would put GKE on a new paragraph to make it stand out more.

File: examples/demo-windows.yaml (hunk 5 above)
Comment (msau42) [reply examples/demo-windows.yaml:20]: From the doc, it looks like a toleration is automatically applied by the admission controller? Do we need the pod to explicitly set it?

File: examples/demo-windows.yaml [hunk 11]
```diff
@@ -0,0 +1,34 @@
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: podpvc-windows
+spec:
+  accessModes:
+    - ReadWriteOnce
+  storageClassName: csi-gce-pd
```
Comment (msau42) [inline examples/demo-windows.yaml:8]: Do we need storageclass with ntfs fstype?

File: README.md (hunk 2 above)
Comment (jingxu97) [reply README.md]: updated

File: examples/demo-windows.yaml (hunk 5 above)
Comment (jingxu97) [reply examples/demo-windows.yaml:20]: I think it will be added in GKE, but not gce cluster.

File: README.md (hunk 7 above)
Comment (jingxu97) [reply README.md:69]: updated, thanks!

File: README.md (hunk 8 above)
Comment (jingxu97) [reply README.md:79]: not yet, I will update it in a separate PR.

File: docs/kubernetes/user-guides/windows.md (hunk 9 above)
Comment (jingxu97) [reply docs/kubernetes/user-guides/windows.md:1]: fixed

File: docs/kubernetes/user-guides/windows.md (hunk 10 above)
Comment (jingxu97) [reply docs/kubernetes/user-guides/windows.md:9]: updated.

File: examples/demo-windows.yaml (hunk 11 above)
Comment (jingxu97) [reply examples/demo-windows.yaml:8]: before we don't need to. but after controller.yaml is changed to have default fsType to ext4, I think we need fstype in storageclass for Windows

**Comment by jingxu97** [comment]
updated, PTAL

**Comment by jingxu97** [comment]
updated. PTAL

File: docs/kubernetes/user-guides/windows.md [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+
+>**Attention:** CSI Windows Beta is only available in the driver version v1.1.0+
+>**Attention:** CSI Windows Alpha is only available in the driver version v1.0.0-v1.0.*
+
+### Install CSI Proxy binary
+CSI proxy can be installed as binary or run as a Windows service on each Windows node. Please see details on CSI Proxy [installation and usage page](https://github.com/kubernetes-csi/csi-proxy/blob/master/README.md#usage).
+
+If you are using kube-up to start a GCE Kubernetes cluster, starting Kubernetes 1.19, [CSI Proxy Alpha](https://github.com/kubernetes-csi/csi-proxy/releases/tag/v0.1.0) binary is automatically installed during node start up. In Kubernetes 1.20, [CSI Proxy Beta.2 v0.2.2](https://github.com/kubernetes-csi/csi-proxy/releases/tag/v0.2.2) is installed and running as a windows service to improve its stablibity.
+
+For GKE cluster, starting from 1.18, [CSI Proxy Beta](https://github.com/kubernetes-csi/csi-proxy/releases/tag/v0.2.2) will be installed automatically.
```
Comment (msau42) [inline docs/kubernetes/user-guides/windows.md:11]: maybe add a note that for GKE, the driver does not need to be manually installed. Instead, they should use the gcloud api.

Comment (jingxu97) [reply docs/kubernetes/user-guides/windows.md:11]: updated.

Comment (msau42) [reply docs/kubernetes/user-guides/windows.md:11]: sorry I don't see the update?

Comment (jingxu97) [reply docs/kubernetes/user-guides/windows.md]: sorry I made this comment too early. now it is updated :)

**Pull request merged**